pkg/new_storage/general.go

```diff
@@ -151,7 +163,7 @@ func (bd *BackupDestination) BackupList() ([]Backup, error) {
 		if err := json.Unmarshal(b, &m); err != nil {
 			result = append(result, Backup{
 				metadata.BackupMetadata{
-					BackupName: strings.Trim(o.Name(), "/"),
+					BackupName: metadataFilePath,
```

Comment:
still strange backup name, I propose return `strings.Trim(o.Name(), "/")` here

Reply:
The core logic to get the backup is through the metadata.json file, right? The name "metadataFilePath" variable here  makes this concept clear and conceals other messy result. If you insist, I can turn this code back.